src/argilla/client/feedback/schemas/remote/records.py

```diff
@@ -169,103 +160,49 @@ def __normalize_suggestions_to_update(
-            )
-            pushed_suggestions.append(
-                RemoteSuggestionSchema.from_api(
-                    payload=pushed_suggestion.parsed,
-                    question_id_to_name={value: key for key, value in self.question_name_to_id.items()},
-                    client=self.client,
-                )
-            )
-
-        self.__dict__["suggestions"] = tuple(pushed_suggestions)
-
     @allowed_for_roles(roles=[UserRole.owner, UserRole.admin])
     def update(
         self,
         suggestions: Optional[
             Union[
-                AllowedSuggestionSchema,
+                SuggestionSchema,
                 Dict[str, Any],
-                List[AllowedSuggestionSchema],
+                List[SuggestionSchema],
                 List[Dict[str, Any]],
             ]
         ] = None,
     ) -> None:
-        """Update a `RemoteFeedbackRecord`. Currently just `suggestions` are supported.
+        """Update a `RemoteFeedbackRecord` in Argilla.
 
         Note that this method will update the record in Argilla directly.
 
         Args:
-            suggestions: can be a single `RemoteSuggestionSchema` or `SuggestionSchema`,
-                a list of `RemoteSuggestionSchema` or `SuggestionSchema`, a single
-                dictionary, or a list of dictionaries. If a dictionary is provided,
-                it will be converted to a `RemoteSuggestionSchema` internally.
+            suggestions: **DEPRECATED** the new list of suggestions to set for the
+                record. It can be a single `SuggestionSchema`, a list of `SuggestionSchema`,
+                a single dictionary or a list of dictionary. If a dictionary is provided,
+                it will be converted to a `SuggestionSchema` internally. Defaults to
+                `None`.
 
         Raises:
             PermissionError: if the user does not have either `owner` or `admin` role.
         """
-        if suggestions:
-            suggestions = self.__normalize_suggestions_to_update(suggestions)
-        else:
-            suggestions = suggestions or [s for s in self.suggestions]
+        if suggestions is not None:
+            warnings.warn(
+                "The `suggestions` argument is deprecated and will be removed in "
+                "future releases. In order to update the suggestions of a record "
+                "assign the `suggestions` attribute directly.",
+                stacklevel=1,
+            )
+
+            self.suggestions = tuple(self.__normalize_suggestions_to_update(suggestions))
 
         self.__updated_record_data()
-        if suggestions:
-            self.__update_suggestions(suggestions=suggestions)
 
     def __updated_record_data(self) -> None:
-        response = records_api_v1.update_record(self.client, self.id, self.to_server_payload())
+        response = records_api_v1.update_record(self.client, self.id, self.to_server_payload(self.question_name_to_id))
```

Comment:
Do we need to pass the `question_name_to_id` as parameter since is already defined at the object level?

Reply:
Yes, the `to_server_payload` method is defined in the `FeedbackRecord` class but the `question_name_to_id` attribute is just available in `RemoteFeedbackRecord`